### PR TITLE
[LPT] Legacy restrictions removal: dequantization operations

### PR DESCRIPTION
**Pull request by eshoguli**
### Details:
 - *Legacy restrictions removal: dequantization operations*

### Tickets:
 - *Task: 61286*
 - *Validation: 62035*

### CODE REVIEW

File: inference-engine/tests/functional/inference_engine/lp_transformations/convert_mul_or_add_finally_transformation_with_dequantization.cpp [hunk 1]
```diff
@@ -110,7 +110,7 @@ std::vector<ConvertMulOrAddFinallyTransformationWithDequantizationTestValues> te
 };
 
 INSTANTIATE_TEST_SUITE_P(
-    smoke_LPT,
+    DISABLED_smoke_LPT,
     ConvertMulOrAddFinallyTransformationWithDequantization,
     ::testing::ValuesIn(testValues),
     ConvertMulOrAddFinallyTransformationWithDequantization::getTestCaseName);
```
[the remark this answers is not in the document]
Comment (eshoguli) [reply inference-engine/tests/functional/inference_engine/lp_transformations/convert_mul_or_add_finally_transformation_with_dequantization.cpp]: done, thanks

File: inference-engine/tests/functional/inference_engine/lp_transformations/mul_add_to_scaleshift_or_power_transformation.cpp [hunk 2]
```diff
@@ -136,7 +136,7 @@ const std::vector<MulAddToScaleshiftOrPowerParams> testValues = {
 };
 
 INSTANTIATE_TEST_SUITE_P(
-    smoke_LPT,
+    DISABLED_smoke_LPT,
     MulAddToScaleshiftOrPowerTransformation,
```
[the remark this answers is not in the document]
Comment (eshoguli) [reply inference-engine/tests/functional/inference_engine/lp_transformations/mul_add_to_scaleshift_or_power_transformation.cpp]: done, thanks

**Review by v-Golubev — APPROVED** [review]

File: inference-engine/src/mkldnn_plugin/mkldnn_plugin.cpp [hunk 3]
```diff
@@ -375,19 +375,6 @@ static void Transformation(CNNNetwork& clonedNetwork, const Config& conf) {
         std::string errMsg;
         return MKLDNNFakeQuantizeNode::isSupportedOperation(node, errMsg);
     });
-    postLPTPassManager.get_pass_config()->set_callback<ngraph::pass::AddMultiplyFusion>([](const_node_ptr &node) -> bool {
-        if (auto mul_op = std::dynamic_pointer_cast<const ngraph::opset1::Multiply>(node)) {
-            auto add_op = std::dynamic_pointer_cast<const ngraph::opset1::Add>(mul_op->get_input_node_shared_ptr(0));
-            auto constant = std::dynamic_pointer_cast<const ngraph::opset1::Constant>(mul_op->get_input_node_shared_ptr(1));
-            bool is_dequantization = mul_op->get_rt_info().count("DEQUANTIZATION") != 0;
-            if (add_op && constant && is_dequantization) {
-                return ngraph::is_type<ngraph::opset1::Convolution>(add_op->get_input_node_shared_ptr(0)) ||
```
Comment (vzinovie) [inline inference-engine/src/mkldnn_plugin/mkldnn_plugin.cpp:384]: Don't we have performance degradations after removing this? Are biases still presence and not swap with dequantization multiplies?

Comment (v-Golubev) [reply inference-engine/src/mkldnn_plugin/mkldnn_plugin.cpp:384]: It's weird, but the AddMultiplyFusion is not even added to postLPTPassManager (we have only ngraph::pass::FakeQuantizeDecomposition and ngraph::pass::UnrollTensorIterator). To my mind, this callback looks like useless.

Comment (vzinovie) [reply inference-engine/src/mkldnn_plugin/mkldnn_plugin.cpp:384]: It's really strange. This callback have to be in part of common optimizations. We need to investigate impact of this lines in common optimizations then.

Comment (eshoguli) [reply inference-engine/src/mkldnn_plugin/mkldnn_plugin.cpp:384]: new tasks was created: 63613

File: inference-engine/tests/ngraph_helpers/lpt_ngraph_functions/include/lpt_ngraph_functions/common/dequantization_operations.hpp [hunk 4]
```diff
@@ -16,7 +16,7 @@ class DequantizationOperations {
     class Convert {
     public:
         Convert();
-        Convert(const ngraph::element::Type outPrecision, const bool addDeqAttr = true);
+        Convert(const ngraph::element::Type outPrecision, const bool toRemove = true);
```
[the remark this answers is not in the document]
Comment (eshoguli) [reply inference-engine/tests/ngraph_helpers/lpt_ngraph_functions/include/lpt_ngraph_functions/common/dequantization_operations.hpp:19]: will be removed in the next PR

**Pull request merged**